forge-game/src/main/java/forge/game/ability/AbilityUtils.java

```diff
@@ -2625,6 +2625,28 @@ public static int xCount(final Card c, final String s, final CardTraitBase ctb)
             return doXMath(game.getStack().getMaxDistinctSources(), expr, c, ctb);
         }
 
+        if (sq[0].equals("MaxSameStoredRolls")) {
+            int max = 0;
+            List<Integer> rolls = c.getStoredRolls();
+            if (rolls != null) {
+                int lastNum = 1;
+                int tally = 0;
+                for (int i = 0; i < rolls.size(); i++) {
```

Comment:
seems like basic for-each would be enough here
also I think it counts wrong in some edge cases (like all rolls being 1)

Reply:
`for (Integer roll : rolls)`